src/lib/libcmd/cat.c

```diff
@@ -489,8 +489,8 @@ int b_cat(int argc, char **argv, Shbltin_t *context) {
             }
         }
         if (sferror(sfstdout)) break;
-        cp = *argv++;
-    } while (cp);
+        ;
+    } while ((cp = *argv++));
```

Comment:
The problem is that this block contains a `continue` that escaped my notice and how it is structured is weird. I've just created a PR that I think is cleaner. Also, the empty statement on the preceding line should be dropped even if this is merged.